modules/tutorials/pages/java-tutorial/retrieving-documents.adoc

```diff
@@ -0,0 +1,130 @@
+:page-pagination:
+
+// suppress inspection "SqlResolve" for whole file
+// suppress inspection "SqlNoDataSourceInspection" for whole file
+= Retrieving documents
+
+:description: pass:q[In this section, you're going to look at two methods of retrieving documents from a collection: you will use the administration console to build simple queries, and also write a short program to retrieve documents matching certain criteria. Both of the methods will introduce `N1QL` (pronounced _nickel_), Couchbase's SQL-based query language.]
+
+[abstract]
+{description}
+
+include::partial$retrieve-documents-using-query-bench-section.adoc[leveloffset="+1"]
+
+== Using the SDK
+
+Of course, you can also retrieve documents using the JDK. In this section, you're going to use the same `N1QL` queries as part of a small Java application. Let's start with a basic record retrieval:
+
+[source, java]
+----
```

Comment:
Will all this Java code be moved into SDK and tested?

Reply:
At some point, yes.